packages/mask/src/components/shared/ApplicationBoard.tsx

```diff
@@ -154,5 +170,89 @@ interface RenderEntryComponentWrapperProps {
 
 function RenderEntryComponentWrapper({ application }: RenderEntryComponentWrapperProps) {
     const RenderEntryComponent = application.entry.RenderEntryComponent!
-    return <RenderEntryComponent disabled={!application.enabled} />
+    return application.entry.nextIdRequired ? (
+        <RenderEntryComponentWithNextIdRequired application={application} />
+    ) : (
+        <RenderEntryComponent disabled={!application.enabled} />
+    )
+}
+
+function RenderEntryComponentWithNextIdRequired({ application }: RenderEntryComponentWrapperProps) {
+    const ui = activatedSocialNetworkUI
+    const { t } = useI18N()
+    const platform = ui.configuration.nextIDConfig?.platform as NextIDPlatform
+    const lastStateRef = currentSetupGuideStatus[ui.networkIdentifier]
+    const lastState_ = useValueRef(lastStateRef)
+    const lastState = useMemo<SetupGuideCrossContextStatus>(() => {
+        try {
+            return JSON.parse(lastState_)
+        } catch {
+            return {}
+        }
+    }, [lastState_])
+    const lastRecognized = useLastRecognizedIdentity()
+    const getUsername = () =>
+        lastState.username || (lastRecognized.identifier.isUnknown ? '' : lastRecognized.identifier.userId)
+    const [username] = useState(getUsername)
+    const personas = useMyPersonas()
+
+    function checkSNSConnectToCurrentPersona(persona: Persona) {
+        return (
+            persona?.linkedProfiles.get(new ProfileIdentifier(ui.networkIdentifier, username))
+                ?.connectionConfirmState === 'confirmed'
+        )
+    }
+
+    const {
+        value = {
+            isNextIdVerify: undefined,
+            isSNSConnectToCurrentPersona: undefined,
+            currentPersonaPublicKey: undefined,
+            currentSNSConnectedPersonaPublicKey: undefined,
+        },
+    } = useAsync(async () => {
+        const currentPersonaIdentifier = await Services.Settings.getCurrentPersonaIdentifier()
+        const currentPersona = (await Services.Identity.queryPersona(currentPersonaIdentifier!)) as Persona
+        const currentSNSConnectedPersona = personas.find((persona) =>
+            checkSNSConnectToCurrentPersona(persona as Persona),
+        )
+        return {
+            isSNSConnectToCurrentPersona: checkSNSConnectToCurrentPersona(currentPersona),
+            isNextIdVerify: await NextIDProof.queryIsBound(currentPersona.publicHexKey ?? '', platform, username),
+            currentPersonaPublicKey: currentPersona
+                ? formatPersonaPublicKey(currentPersona.fingerprint ?? '', 4)
```

Comment:
Please don't do this, I suggest you use the formatter function in jsx.

Reply:
done